src/tools/interop/idt/capture/platform/android/android.py

```diff
@@ -0,0 +1,220 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import ipaddress
+import os
+import time
+import typing
+
+from capture.base import PlatformLogStreamer
+from capture.file_utils import create_standard_log_name
+from capture.shell_utils import BashRunner
+
+
+class Android(PlatformLogStreamer):
+    """
+    Class that supports:
```

Comment:
Can we add support to pull the btsnoop logs? The link key can also be valuable. Here are my notes

```
Android

The information below was derived while testing a rooted OnePlus 7T device, running Pixel Experience (Android 13) custom ROM.

Without root access, you will be unable to directly access the /data/misc/bluetooth/ location.

The HCI Snoop logs are available @ /data/misc/bluetooth/logs/btsnoop_hci.log
After a Bluetooth connection has been established with a device, the link key can be accessed (on a rooted phone) @ /data/misc/bluedroid/bt_config.conf
The link key can be used with various analyzer to support decoding encrypted data channels. https://www.ellisys.com/technology/een_bt09.pdf
Enabling Bluetooth HCI logging

    Settings → System → Developer Options → USB debugging: Enabled
    Settings → System → Developer Options → Enable Bluetooth HCI snoop log: Enabled
    Settings → System → Developer Options → Logger buffer sizes: Select the maximum value

Rebooting the device, or toggling Bluetooth, might be required after toggling the setting.
Pulling the BT Snoop Logs (rooted phone)

    Copy the file to a location that can be accessed without root.
    adb shell 'su -c cp /data/misc/bluetooth/logs/btsnoop_hci.log /sdcard'

    Pull the file to local host.
    adb pull /sdcard/btsnoop_hci.log

Pulling the BT Snoop Logs (without rooted phone)

    Trigger bug report.
    adb bugreport
    Extract the bug report contents. Access the btsnoop_hci.log @ FS/data/misc/bluetooth/logs/btsnoop_hci.log


```

Reply:
Resolving duplicate comment.